server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -197,12 +233,62 @@ public void testSuccessfulSnapshot() {
+                                                        ),
+                                                        assertNoFailureListener(r -> {
+                                                            assertEquals(
+                                                                (long) documents,
+                                                                Objects.requireNonNull(r.getHits().getTotalHits()).value
+                                                            );
+                                                            documentCountVerified.set(true);
+                                                        }));
+                                                })
+                                            )));
+                                    }));
+                                final AtomicInteger countdown = new AtomicInteger(documents);
+                                masterNode.client.admin().indices().putMapping(
+                                    new PutMappingRequest(index).type("_doc").source("foo", "type=text"),
+                                    assertNoFailureListener(r -> {
+                                            for (int i = 0; i < documents; ++i) {
+                                                masterNode.client.bulk(
+                                                    new BulkRequest().add(new IndexRequest(index).source(
+                                                        Collections.singletonMap("foo", "bar" + i)))
+                                                        .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE),
+                                                    assertNoFailureListener(
+                                                        bulkResponse -> {
+                                                            assertFalse(
+                                                                "Failures in bulkresponse: " + bulkResponse.buildFailureMessage(),
+                                                                bulkResponse.hasFailures());
+                                                            if (countdown.decrementAndGet() == 0) {
+                                                                afterIndexing.run();
+                                                            }
+                                                        }));
+                                            }
+                                            if (documents == 0) {
+                                                afterIndexing.run();
+                                            }
+                                        }
+                                    ));
+                            }))));
+        runUntil(
+            () -> createdSnapshot.get() && snapshotRestored.get() && documentCountVerified.get(), TimeUnit.MINUTES.toMillis(5L));
```

Comment:
Why does deterministicTaskQueue.runAllRunnableTasks();	 no longer work for us?

Reply:
@andrershov because it only runs all the tasks that should execute at a certain point in time. It does not itself advance the internal (fake) clock to run tasks scheduled for a later time.
For tests that don't depend on any timeouts or so and just run a series of callbacks that's fine, but for tests that have node failures or the like that have rely on `.schedule(` calls for timeouts or similar we need to advance the clock in a look which is what this method does.